File: doctorClinic/doctor_clinic_document.py
class DoctorClinicDocument:
    def __init__(self):
        self.doctor = ''
        self.clinic = ''
        self.number = ''
        self.address = ''
        self.desc = ''
    
    def setDataByIndex(self, index, data):
        if index == 0:
            self.doctor = data
            print("set doctor name complete!")
        elif index == 1:
            self.clinic = data
            print("set doctor name complete!")
        elif index == 2:
            self.number = data
            print("set doctor name complete!")
        elif index == 3:
            self.address = data
            print("set doctor name complete!")
        else:
            self.desc += data
            print("set desc name complete!")

    def getDataToDict(self):
        return {'doctor' : self.doctor, 
                 'clinic' : self.clinic, 
                 'number': self.number, 
                 'address' : self.address,
                 'desc' : self.desc}

    def resetData(self):
        self.doctor = ''
        self.clinic = ''
        self.number = ''
        self.address = ''
        self.desc = ''

File: dart/lib/src/doctorClinic/doctor_clinic_document.dart

class DoctorClinicDocument {
  String doctor = '';
  String clinic = '';
  String number = '';
  String address = '';
  String desc = '';

  DoctorClinicDocument();

  void resetData() {
    doctor = '';
    clinic = '';
    number = '';
    address = '';
    desc = '';
  }

  Map<String, String> getDataToDict() {
    return {'doctor' : doctor,
      'clinic' : clinic,
      'number': number,
      'address' : address,
      'desc' : desc};
  }

  void setDataByIndex(int index, String data) {
    switch(index) {
      case 0:
        doctor = data;
        break;
      case 1:
        clinic = data;
        break;
      case 2:
        number = data;
        break;
      case 3:
        address = data;
        break;
      default:
        desc += data;
        break;
    }
  }
}
